cpu/cortexm_common/cortexm_init.c

```diff
@@ -72,3 +72,41 @@ void cortexm_init(void)
     cortexm_init_isr_priorities();
     cortexm_init_misc();
 }
+
+static const uint32_t BFARVALID_MASK = (0x80 << SCB_CFSR_BUSFAULTSR_Pos);
+
+bool cpu_check_address(volatile const char *address)
+{
+#if defined(CPU_ARCH_CORTEX_M3) || defined(CPU_ARCH_CORTEX_M4) || \
+    defined(CPU_ARCH_CORTEX_M4F) || defined(CPU_ARCH_CORTEX_M7)
+    bool is_valid = true;
+
+    /* Clear BFARVALID flag */
+    SCB->CFSR |= BFARVALID_MASK;
+
+    /* Ignore BusFault by enabling BFHFNMIGN and disabling interrupts */
+    uint32_t mask = __get_FAULTMASK();
+    __disable_fault_irq();
+    SCB->CCR |= SCB_CCR_BFHFNMIGN_Msk;
+
+    *address;
+    /* Check BFARVALID flag */
+    if ((SCB->CFSR & BFARVALID_MASK) != 0)
+    {
+        /* Bus Fault occured reading the address */
+        is_valid = false;
+    }
+
+    /* Reenable BusFault by clearing  BFHFNMIGN */
+    SCB->CCR &= ~SCB_CCR_BFHFNMIGN_Msk;
+    __set_FAULTMASK(mask);
+
+    return is_valid;
+#else
+    /* Cortex-M0 doesn't have BusFault */
+    (void) address;
+
+    printf("[warning] %s: Cortex-M0 doesn't have BusFault\n", __func__);
+    return true;
```

Comment:
Is there a use-case for having this compiled in on Cortex-M0? If not, I suggest guarding the whole function.

(returning true seems wrong)

Reply:
Or wouldn't it be possible to catch an "expected" hardfault and that way somehow replicate the functionality on M0?

Reply:
@kaspar030 Yes, just wrote a catcher for M0.